interview-questions.md

```diff
@@ -8,50 +8,50 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is the input that is required by the function. Think of a parameter as a placeholder for the data being passed into the function, while the argument is the actual data that gets passed into the function. A parameter is what goes into the function while the argument is the data that you call the function with.
 
-Researched answer:
+Researched answer: Parameters are variables that will be passed to the function for processing. Arguments are values passed into the function when it is invoked.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: .map() takes in the index, the value, and the array used with the higher order function. The value is required while the index and the array are optional.
 
-Researched answer:
+Researched answer: The built in method .map() has three built in parameters. Because .map is only used when working with arrays, the three parameters are the value of the array, the index of the array, and the array that the higher order function was called on. The array, and the index of the array at its current iteration are optional. The value of the array at its current iteration is always required.
 
-3. What is the difference between map and filt er?
+3. What is the difference between map and filter?
 
-Your answer:
+Your answer: map will change each item in the array usually through iteration and return an array of the same size. filter will filter through the array and keep what we do want and get rid of what we dont want.
 
-Researched answer:
+Researched answer: .map will modify each item in an array and return an array of the SAME size. filter will look at each item in the array and return a subset of the original array.
 
```

Comment:
more content can be comparing your thoughts of these topics when you first saw by how you felt after using them